io/snmp/snmp.js

```diff
@@ -51,6 +51,53 @@ module.exports = function(RED) {
     }
     RED.nodes.registerType("snmp",SnmpNode);
 
+ function SnmpSNode(n) {
+        RED.nodes.createNode(this, n);
+        this.community = n.community;
+        this.host = n.host;
+        this.version = (n.version === "2c") ? snmp.Version2c : snmp.Version1;
+        var node = this;
+
+        this.on("input", function (msg) {
+            var host = node.host || msg.host;
+            var community = node.community || msg.community;
+            var oids = msg.oids;
+            node.log(oids + ' ' + host + ' ' + msg.host+ community + msg.community);
+            if (oids) {
+                node.session = snmp.createSession(host, community, { version: node.version });
+                node.session.set(oids, function (error, varbinds) {
+                    if (error) {
+                        node.error(error.toString(), msg);
+                    }
+                    else {
+                        for (var i = 0; i < varbinds.length; i++) {
+                            if (snmp.isVarbindError(varbinds[i])) {
+                                node.error(snmp.varbindError(varbinds[i]), msg);
+                            }
+                            else {
+                                if (varbinds[i].type == 4) { varbinds[i].value = varbinds[i].value.toString(); }
+                                varbinds[i].tstr = snmp.ObjectType[varbinds[i].type];
+                                //node.log(varbinds[i].oid + "|" + varbinds[i].tstr + "|" + varbinds[i].value);
+                            }
+                        }
+                        msg.oid = oids;
+                        msg.payload = varbinds;
+                        node.send(msg);
```

Comment:
I think @dceejay's point is that we output nodes don't generally output anything back into the flow.
Based on the [documentation](https://github.com/stephenwvickers/node-net-snmp/blob/cc4860b1b244b9822979d9822056a30e1be6cbb4/README.md), I'd modify their sample:

```javascript
var varbinds = [
    {
        oid: "1.3.6.1.2.1.1.5.0",
        type: snmp.ObjectType.OctetString,
        value: "host1"
    }, {
        oid: "1.3.6.1.2.1.1.6.0",
        type: snmp.ObjectType.OctetString,
        value: "somewhere"
    }
];

session.set (varbinds, function (error, varbinds) {
    if (error) {
        console.error (error.toString ());
    } else {
        for (var i = 0; i < varbinds.length; i++) {
            // for version 1 we can assume all OIDs were successful
            console.log (varbinds[i].oid + "|" + varbinds[i].value);
        
            // for version 2c we must check each OID for an error condition
            if (snmp.isVarbindError (varbinds[i]))
                console.error (snmp.varbindError (varbinds[i]));
            else
                console.log (varbinds[i].oid + "|" + varbinds[i].value);
        }
    }
});
```
to something like this:

```javascript
session.set (varbinds, function (error, varbinds) {
    if (error) {
        node.error (error.toString(), msg);
    } else {
        for (var i = 0; i < varbinds.length; i++) {
            // for version 1 we can assume all OIDs were successful
            // Don't do anything with this since unless there was an error, we assume success
            // console.log (varbinds[i].oid + "|" + varbinds[i].value);
        
            // for version 2c we must check each OID for an error condition
            if (snmp.isVarbindError (varbinds[i]))
                node.error (snmp.varbindError (varbinds[i]), msg);
            // Don't do anything with this since unless there was an error, we assume success
            //else
            //    console.log (varbinds[i].oid + "|" + varbinds[i].value);
        }
    }
});
```

There is no need to `node.send(msg)`, since there is no new information to be sent back into the flow.

What do you propose as the method for a user to set the varbinds? Would you want to write a helper inside of the node to convert the type `"Integer"` to `snmp.ObjectType.Integer` and `"Gauge32"` to `snmp.ObjectType.Gauge32`, etc?